components/places/src/api/places_api.rs

```diff
@@ -310,6 +318,57 @@ impl PlacesApi {
         Ok(sync_ping)
     }
 
+    // This is the new sync API until the sync manager lands. It's currently
+    // not wired up via the FFI - it's possible we'll do declined engines too
+    // before we do.
+    pub fn sync(
+        &self,
+        client_init: &sync15::Sync15StorageClientInit,
+        key_bundle: &sync15::KeyBundle,
+    ) -> Result<(ServiceStatus, telemetry::SyncTelemetryPing)> {
+        let mut guard = self.sync_state.lock().unwrap();
+        let conn = self.open_sync_connection()?;
+        if guard.is_none() {
+            *guard = Some(SyncState {
+                mem_cached_state: Cell::default(),
+                disk_cached_state: Cell::new(self.get_disk_persisted_state(&conn)?),
+            });
+        }
+
+        let mut service_status = ServiceStatus::Ok;
+        let sync_state = guard.as_ref().unwrap();
+        // Note that counter-intuitively, this must be called before we do a
+        // bookmark sync too, to ensure the shared global state is correct.
+        HistoryStore::migrate_v1_global_state(&conn)?;
```

Comment:
Do we need to call this after we sync the history store, too, or is just once before syncing both enough?

Reply:
Just once before any syncing is fine.